staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1176,21 +1247,42 @@ type cacheWatcher struct {
 	// human readable identifier that helps assigning cacheWatcher
 	// instance with request
 	identifier string
+
+	// drainInputBuffer indicates whether we should delay closing this watcher
+	// and send all event in the input buffer.
+	drainInputBuffer bool
+
+	// bookmarkAfterResourceVersion holds an RV that indicates when we should start delivering bookmark events
+	// if this field holds the value of 0 that means we don't have any special preferences toward delivering bookmark events
+	bookmarkAfterResourceVersion uint64
```

Comment:
Let's maybe state explicitly that this field is assumed to be immutable for the entire life of cacheWatcher.